### PR TITLE
Support lookup extr dims filter query

### PR DESCRIPTION
**Pull request by srggrs**
This PR fix:
* support for global lookup in the filter menu and query
* support for lookup map in the filter menu
* support for extraction function in the filter menu and query

The PyDruid requirement has been changed to reflect inclusion of Registered Lookup class as in [Registered lookups #153](https://github.com/druid-io/pydruid/pull/153)

Also there is quite a bit of code for issuing the Druid query, (e.g. using "Phase 1" and "Phase 2" queries). Can someone explain to me the reasons behind that? I can help in cleaning up and refactoring

Would fix this #5973

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -49,7 +49,8 @@ pathlib2==2.3.0
 polyline==1.3.2
 py==1.7.0                 # via retry
 pycparser==2.19           # via cffi
-pydruid==0.5.0
+#pydruid==0.5.0
+git+https://github.com/switchdin/pydruid.git@registered-lookups # pydruid
```
Comment (mistercrunch) [inline requirements.txt]: We can't do this as it's a security issue. Get your feature merged / released on `pydruid` first.

**Review by mistercrunch — CHANGES_REQUESTED** [review]

File: superset/migrations/versions/bc649e169f14_support_for_lookup_extraction_filters_.py [hunk 2]
```diff
@@ -0,0 +1,670 @@
...
+revision = 'bc649e169f14'
+down_revision = 'c82ee8a39623'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_table('long_lat')
```
Comment (mistercrunch) [inline superset/migrations/versions/bc649e169f14_support_for_lookup_extraction_filters_.py]: this migration doesn't look right, it's doing unrelated operations

Comment (srggrs) [reply superset/migrations/versions/bc649e169f14_support_for_lookup_extraction_filters_.py]: yeah sorry my bad. This migration is not needed at all! I removed it

File: requirements.txt (hunk 1 above)
Comment (srggrs) [reply requirements.txt]: okay, fixed! thx

File: requirements.txt [hunk 3]
```diff
@@ -49,7 +49,7 @@ pathlib2==2.3.0
 polyline==1.3.2
 py==1.7.0                 # via retry
 pycparser==2.19           # via cffi
-pydruid==0.5.0
+pydruid==0.5.2
```
Comment (mistercrunch) [inline requirements.txt]: You have to bump the range in `setup.py` as well

File: superset/connectors/druid/models.py [hunk 4]
```diff
@@ -33,7 +33,10 @@
 import pandas
 from pydruid.client import PyDruid
 from pydruid.utils.aggregators import count
-from pydruid.utils.dimensions import MapLookupExtraction, RegexExtraction
+from pydruid.utils.dimensions import (
+    MapLookupExtraction, RegexExtraction,
+    TimeFormatExtraction, RegisteredLookupExtraction
```
Comment (mistercrunch) [inline superset/connectors/druid/models.py]: trailing comma!

File: superset/connectors/druid/models.py [hunk 5]
```diff
@@ -859,6 +862,12 @@ def values_for_column(self,
             'Getting values for columns [{}] limited to [{}]'
             .format(column_name, limit))
         # TODO: Use Lexicographic TopNMetricSpec once supported by PyDruid
+        dim_qry = column_name
+        if self.get_column(column_name).dimension_spec and\
```
Comment (mistercrunch) [inline superset/connectors/druid/models.py]: let's call `self.get_column(column_name)` only once: `col = self.get_column(column_name)`

Comment (mistercrunch) [reply superset/connectors/druid/models.py]: indent isn't right
```python
if (
        self.get_column(column_name).dimension_spec and
        ...
):
    ...
```

Comment (srggrs) [reply superset/connectors/druid/models.py]: I fix indentation as per flake8 but it looks weird...

File: requirements.txt (hunk 1 above)
Comment (nosahama) [reply requirements.txt]: @srggrs still no release yet?

**Pull request closed without merging**